network/p2p/middleware.go

```diff
@@ -313,6 +313,8 @@ func (m *Middleware) stop() {
 // Dispatch should be used whenever guaranteed delivery to a specific target is required. Otherwise, Publish is
 // a more efficient candidate.
 func (m *Middleware) SendDirect(msg *message.Message, targetID flow.Identifier) error {
+	m.metrics.DirectMessageStarted(msg.ChannelID)
+
```

Comment:
```suggestion
	m.metrics.DirectMessageStarted(msg.ChannelID)
	defer m.metrics.DirectMessageFinished(msg.ChannelID)

```

Otherwise errors will cause the metric to begin reporting incorrectly